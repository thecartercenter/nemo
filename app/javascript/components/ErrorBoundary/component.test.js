import React from 'react';
import { mount } from 'enzyme';

import Component from './component';

const mockProps = {
  message: 'message',
  children: 'children',
};

const DangerousComponent = () => {
  throw new Error();
};

it('renders (default)', () => {
  const wrapper = mount(<Component {...mockProps} />);
  expect(wrapper).toMatchSnapshot();
});

it('renders (error)', () => {
  const wrapper = mount(
    <Component {...mockProps}>
      <DangerousComponent />
    </Component>,
  );
  expect(wrapper).toMatchSnapshot();
});
