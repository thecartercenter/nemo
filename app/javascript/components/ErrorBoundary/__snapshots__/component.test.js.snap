// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders (default) 1`] = `
<ErrorBoundary
  message="message"
>
  <ErrorBoundary
    fallback={
      <ErrorFallback
        message="message"
      />
    }
  >
    children
  </ErrorBoundary>
</ErrorBoundary>
`;

exports[`renders (error) 1`] = `
<ErrorBoundary
  message="message"
>
  <ErrorBoundary
    fallback={
      <ErrorFallback
        message="message"
      />
    }
  >
    <ErrorFallback
      message="message"
    >
      <div
        className="alert alert-danger"
        role="alert"
      >
        message
      </div>
    </ErrorFallback>
  </ErrorBoundary>
</ErrorBoundary>
`;
